terraform/lib/dependabot/terraform/file_updater.rb

```diff
@@ -21,11 +22,15 @@ def updated_dependency_files
           next unless file_changed?(file)
 
           updated_content = updated_terraform_file_content(file)
-          raise "Content didn't change!" if updated_content == file.content
+
+          raise "Content didn't change!" unless updated_content != file.content
 
           updated_files << updated_file(file: file, content: updated_content)
         end
 
+        updated_files << update_lockfile_declaration 
```

Comment:
Nice cleanup! I'm still wondering if this should return the content instead of the `updated_file(file..)`? That way you could only append it to `updated_files` if the content has changed. 

Something like

```
updated_lockfile_content = update_lockfile_declaration
if (updated_lockfile_content && lock_file.content != update_lockfile_content)
  updated_files << updated_file(file: lock_file, content: update_lockfile_content)
end
```

Reply:
It would be good to add a test case to make sure `updated_dependency_files` doesn't return a lockfile if it doesn't change.

Reply:
Implemented @feelepxyz's  suggestion and created a test/fixture to make sure `updated_dependency_files` doesn't return a lockfile if it doesn't change.

The test uses a terraform project fixture that is fully up-to-date. However, since no files change after running dependabot on the fixture, it raises a "No files changed!" error. 

The error makes the function quit early and prevents the test from directly checking whether the lockfile is unmodified or appended to the `updated_files` array.

So instead, I am testing for the error, which indicates that the lockfile didn't change.